kv/kvserver: TestClosedTimestampFrozenAfterSubsumption failed

kv/kvserver.TestClosedTimestampFrozenAfterSubsumption [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20478739?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20478739?buildTab=artifacts#/) on master @ [7d65e388382ec2081f3604d8bed2bbaab9ac706a](https://github.com/cockroachdb/cockroach/commits/7d65e388382ec2081f3604d8bed2bbaab9ac706a):


```
=== RUN   TestClosedTimestampFrozenAfterSubsumption
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/15f2e1706383de9d6af3292f657b4304/logTestClosedTimestampFrozenAfterSubsumption48254774
    test_log_scope.go:82: use -show-logs to present logs inline
    closed_timestamp_test.go:850: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15f2e1706383de9d6af3292f657b4304/logTestClosedTimestampFrozenAfterSubsumption48254774
--- FAIL: TestClosedTimestampFrozenAfterSubsumption (103.93s)
=== RUN   TestClosedTimestampFrozenAfterSubsumption/rhs_lease_transfer_while_subsumed
    testcluster.go:1414: condition failed to evaluate within 1m30s: from testcluster.go:1416: leader hasn't stepped down yet
--- FAIL: TestClosedTimestampFrozenAfterSubsumption/rhs_lease_transfer_while_subsumed (99.26s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152105 kv/kvserver: TestClosedTimestampFrozenAfterSubsumption failed [A-testing C-bug C-investigation C-test-failure O-robot P-2 T-kv branch-release-25.3]
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClosedTimestampFrozenAfterSubsumption.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54554

--- a/pkg/kv/kvserver/closed_timestamp_test.go
+++ b/pkg/kv/kvserver/closed_timestamp_test.go
@@ -624,8 +624,11 @@ func TestClosedTimestampFrozenAfterSubsumption(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.UnderRace(t)
-	skip.UnderDeadlock(t)
+	skip.UnderDuress(t)
+
+	// Increase the verbosity of the logs to help debug the test if it fails, especially
+	// raft related logs when the test tries to transfer the lease non-cooperatively.
+	require.NoError(t, log.SetVModule("raft=4,*=1"))
 
 	for _, test := range []struct {
 		name string

--- a/pkg/testutils/testcluster/testcluster.go
+++ b/pkg/testutils/testcluster/testcluster.go
@@ -1267,6 +1267,7 @@ func (tc *TestCluster) MoveRangeLeaseNonCooperatively(
 		return nil, errors.Errorf("must set StoreTestingKnobs.AllowLeaseRequestProposalsWhenNotLeader")
 	}
 
+	log.Dev.Infof(ctx, "moving lease non-cooperatively of range %v to %v", rangeDesc, dest)
 	destServer, err := tc.FindMemberServer(dest.StoreID)
 	if err != nil {
 		return nil, err
@@ -1380,8 +1381,17 @@ func (tc *TestCluster) ensureLeaderStepsDown(
 				leaderStore = curStore
 				leaderNode = s
 				leaderReplica = curR
+
+				// Make sure that the leader is fortified because in the next step we
+				// will stop store liveness messages to the leader, and we want to cause
+				// it to step down. If the leader isn't fortified yet, stopping store
+				// liveness messages to it will not cause it to step down.
+				if curR.RaftStatus().LeadSupportUntil.IsEmpty() {
+					return errors.Errorf("leader is not fortified")
+				}
 			}
 		}
+
 		// At this point we have iterated over all nodes in the cluster, if we
 		// haven't found a leader, wait for a bit for one to step up.
 		if leaderStore == nil {

